### ISSUE TITLE
Prepare and deploy on NPM

### ISSUE THREAD

**Issue by HexaField** [XRFoundation/XREngine#2844]
A few things need to be fixed up before we can use xrengine on npm

- fix cross-dependency (example: @xrengine/engine requiring @xrengine/client-core)
- fix monorepo direct imports (example: '../../../../../engine/src/...')
- fix build scripts
- fix publish script

**Comment by HexaField** [comment]
Closing in favour of changes to how we are thinking about the stack. Instead of publishing everything to NPM, we are opting the project API.
